internal/yggdrasil/configuration-assembler.go

```diff
@@ -467,8 +473,26 @@ func (a *configurationAssembler) getMetricsReceiverConfiguration(device *v1alpha
 				result.RequestNumSamples = receiverConfig.RequestNumSamples
 			}
 			result.URL = receiverConfig.URL
+
+			if result.URL != "" && receiverConfig.CaSecretName != "" {
```

Comment:
Do you think there should be a check that the scheme is https?

Reply:
is it really that important? what if there's no scheme in URL? URL can be just host. That is if we want to be really strict and accurate.

Reply:
I agree with @masayag